src/utils/i18n.ts

```diff
@@ -17,20 +53,19 @@ export const getSupportedLocale = (usedLocale: string) => {
 export function detectLocale(): SupportedLocale {
   if (typeof window === 'undefined') {
     /**
-     * Server locale detection. This requires e.g LANG environment variable to be set on the server.
+     * Server locale detection. This requires e.g NMP_BRAND environment variable to be set on the server.
      */
-    const serverLocale =
-      process.env.NMP_LANGUAGE ||
-      Intl.DateTimeFormat().resolvedOptions().locale;
+    const serverLocale = detectByBrand();
     return getSupportedLocale(serverLocale);
   }
 
   try {
     /**
      * Client locale detection. Expects the lang attribute to be defined.
      */
-    const htmlLocale = document.documentElement.lang;
-    return getSupportedLocale(htmlLocale);
+    const htmlLocale = document?.documentElement?.lang;
+    const hostLocale = detectByHost();
+    return getSupportedLocale(htmlLocale ?? hostLocale);
```

Comment:
For this fallback approach to work in most cases, you would need to validate/parse document.documentElement.lang since in all cases I've seen, the value is set, just not always correctly. `no` is a typical invalid value. Ideally we'd use a library for this but thats extra kb we could do without. 
You could add a detectByLangAttribute() function that checks for valid en, fi, nb, sv and da values perhaps?

Reply:
So I have added a fallback which if the html lang is incorrect, fallback to detection using hostname.

Reply:
Console warning when the detection fails:
<img width="593" alt="Screenshot 2024-01-10 at 10 50 36" src="https://github.com/warp-ds/icons/assets/21952706/ea0cbba3-5ef0-4d47-8c8c-2c1e5b6f8ade">

Reply:
What about something like this?
```
Unsupported "${htmlLocale}" locale set in lang attribute on the html tag.
Falling back to locale based on hostname.

Reply:
Printing `htmlLocale` isn't really useful since someone checking this on console can easily see what the html lang is set to. Wordings I'll tweak a bit 👍